### PR TITLE
externals: fix find Crypto++

### PR DESCRIPTION
**Pull request by SuperSamus**
With this, Citra correctly finds Crypto++ on my system (NixOS).

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7189)

### CODE REVIEW

**Comment by Steveice10** [comment]
@Castor215 Can you check this one?

**Comment by Castor215** [comment]
> @Castor215 Can you check this one?

Still not working for me. It needs to search for both libcrypto++ and libcryptopp.

File: externals/cmake-modules/Findcryptopp.cmake [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
     find_path(CRYPTOPP_INCLUDE_DIRS NAMES cryptlib.h
             PATHS
-            ${CRYPTOPP_TMP_INCLUDE_DIRS}
+            ${CRYPTOPP_TMP_INCLUDE_DIRS}/cryptopp
             /usr/include
-            /usr/include/crypto++
+            /usr/include/cryptopp
             /usr/local/include
-            /usr/local/include/crypto++
+            /usr/local/include/cryptopp
```
Comment (Steveice10) [inline externals/cmake-modules/Findcryptopp.cmake:10]: I believe these paths should look for both crypto++ and cryptopp

File: externals/cmake-modules/Findcryptopp.cmake [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
             )
 
-    find_library(CRYPTOPP_LIBRARY_DIRS NAMES crypto++
+    find_library(CRYPTOPP_LIBRARY_DIRS NAMES cryptopp
```
Comment (Steveice10) [inline externals/cmake-modules/Findcryptopp.cmake:13]: Believe this should look for both crypto++ and cryptopp as well

**Comment by SuperSamus** [comment]
Are you sure about this whole `++` thing?
The library generated by the Makefile [is named `libcryptopp`](https://github.com/weidai11/cryptopp/blob/master/GNUmakefile#L19), the [include dir is `cryptopp`](https://github.com/weidai11/cryptopp/blob/master/GNUmakefile#L1440).
@Castor215 what distro are you on?

**Comment by Castor215** [comment]
> Are you sure about this whole `++` thing? The library generated by the Makefile [is named `libcryptopp`](https://github.com/weidai11/cryptopp/blob/master/GNUmakefile#L19), the [include dir is `cryptopp`](https://github.com/weidai11/cryptopp/blob/master/GNUmakefile#L1440). @Castor215 what distro are you on?

Debian (testing/trixie). [As you can see](https://packages.debian.org/trixie/amd64/libcrypto++-dev/filelist), the -dev package contains a libcrypto++.pc file, but no libcryptopp.pc file. As for [the library package](https://packages.debian.org/trixie/amd64/libcrypto++8/filelist), it contains both libcrypto++.so and libcryptopp.so (which is a symlink to libcrypto++.so).

**Comment by SuperSamus** [comment]
And the reason the name is different is... [Debian changed it](https://sources.debian.org/patches/libcrypto%2B%2B/8.9.0-1/fix_library_name.patch/). (Why?)

**Comment by Steveice10** [comment]
> Debian (testing/trixie). [As you can see](https://packages.debian.org/trixie/amd64/libcrypto++-dev/filelist), the -dev package contains a libcrypto++.pc file, but no libcryptopp.pc file. As for [the library package](https://packages.debian.org/trixie/amd64/libcrypto++8/filelist), it contains both libcrypto++.so and libcryptopp.so (which is a symlink to libcrypto++.so).

So does this need the other changes I commented or does it work as-is with just the search clause?

**Comment by Castor215** [comment]
> > Debian (testing/trixie). [As you can see](https://packages.debian.org/trixie/amd64/libcrypto++-dev/filelist), the -dev package contains a libcrypto++.pc file, but no libcryptopp.pc file. As for [the library package](https://packages.debian.org/trixie/amd64/libcrypto++8/filelist), it contains both libcrypto++.so and libcryptopp.so (which is a symlink to libcrypto++.so).
> 
> So does this need the other changes I commented or does it work as-is with just the search clause?

It seems to find it on mine with just the search clause.

**Comment by SuperSamus** [comment]
I'm still adding `crypto++` search paths just in case.

**Review by Steveice10 — APPROVED** [review]
LGTM if @Castor215 can just confirm the final product still works on Debian.

**Comment by Castor215** [comment]
Works for me. Thank you for doing this!

**Pull request merged**